templates/azml-app/app/generateScore.py

```diff
@@ -0,0 +1,111 @@
+import os
+
+SCORE_FILE_TEMPLATE = """from fastapi import Request
+from transformers import TOKENIZER_TYPE, MODEL_TYPE
+IMPORT_PIPELINE_TYPE
+import os
+
+def init():
+    pipe = None
+    model = None
+    tokenizer = None
+    print("Loading model...")
+    model = MODEL_TYPE.from_pretrained(f"SAFETENSOR_PATH", local_files_only=True)
+    print("Loading tokenizer...")
+    tokenizer = TOKENIZER_TYPE.from_pretrained(f"TOKENIZER_PATH", local_files_only=True)
+    PIPELINE_LOADING
+    return pipe, model, tokenizer
+
+async def run(request, pipe, tokenizer, model):
+    if pipe is not None:
+        body = await request.json()
+        prompt = body.get("prompt", "")
+        output = pipe(prompt, max_new_tokens=100)
+        result = output[0]["generated_text"]
+        return {"response": result}
+    if model is not None and tokenizer is not None:
+        body = await request.json()
+        prompt = body.get("prompt", "")
+        inputs = tokenizer(prompt, return_tensors="pt")
+        output = model.generate(**inputs, max_new_tokens=100)
+        result = tokenizer.decode(output[0], skip_special_tokens=True)
+        return {"response": result}
+    else:
+        print("Error: Model not loaded yet")
+        return {"error": "Model not loaded yet"}
+
+"""
+IMPORT_PIPELINE_TYPE = """from transformers import PIPELINE_TYPE"""
+PIPELINE_LOADING_TEMPLATE = """print("Constructing pipeline...")
+    pipe = PIPELINE_TYPE(TASK_NAMEmodel=model, tokenizer=tokenizer)"""
+KNOWN_TASK_PIPELINE_MAP = {
+    "chat-completion": "TextGenerationPipeline",
+}
+
+
+class ScoreFileGenerator:
+    def __init__(self, model_dir_name: str = None, pipeline_class_name: str = None, tokenizer_class_name: str = None,
+                 model_loader_class_name: str = None, pipeline_task_name: str = None, tokenizer_path: str = None,
+                 model_asset_path: str = None):
+        """
+            Generates a score file for the model.
+            :param model_dir_name: The name of the model directory.
+            :param pipeline_type_name: The type of the pipeline.
+            :param tokenizer_type_name: The type of the tokenizer.
+            :param model_loader_type_name: The type of the model loader.
+            :param pipeline_task_name: The task name for the pipeline.
+            :param tokenizer_path: The path to the tokenizer.
+            :param model_asset_path: The path to the safetensor files.
+        """
+        self.model_dir_name = model_dir_name
+        self.pipeline_class_name = pipeline_class_name
+        self.tokenizer_class_name = tokenizer_class_name
+        self.model_loader_class_name = model_loader_class_name
+        self.pipeline_task_name = pipeline_task_name
+        self.tokenizer_path = tokenizer_path
+        self.model_asset_path = model_asset_path
+
+        self.score_file_type = None
+
+    def generate(self):
+        print("Generating score file...")
+        if self.score_file_type.lower() == "mlflow":

```

Comment:
>score_file_type [](http://example.com/codeflow?start=16&length=15)

if this is None, then this will throw attributeerror on .lower(). replace with if self.score_file_type and self.score_file_type.lower() == "mlflow":